include/secp256k1_thresholdsig.h

```diff
@@ -0,0 +1,147 @@
+ *      pubcoeff: array of k EC points used by signers to verify their shards (cannot be NULL)
+ *  In:   seckey: unmodified secret key (cannot be NULL)
+ *             k: signing threshold; size of `pubcoeff` array
+ *             n: number of signers; size of `shards` array
+ */
+SECP256K1_API int secp256k1_thresholdsig_keysplit(
+    const secp256k1_context *ctx,
+    secp256k1_thresholdsig_keyshard *shards,
+    secp256k1_pubkey *pubcoeff,
+    const unsigned char *seckey,
+    const size_t k,
+    const size_t n
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(4);
+
+/** Verifies a keyshard against a set of public coefficients and updates the set of signing keys
+ *
+ *  This function serves two very different purposes. First, if `privshard` is non-NULL, it
+ *  verifies that this shard is consistent with a set of public coefficients which were output
+ *  from `secp256k1_thresholdsig_keysplit`. If `seckey` is non-NULL, it updates this value to
+ *  become a sum of all other signers' private shards.
+ *
+ *  Secondly, it computes the public version of every other signer's shard, and updates the
+ *  array `signer_pubkeys` to become an array of all other signers' public keys.
+ *
+ *  The resulting keys, when weighted by appropriate Lagrange coefficients, will sum to the MuSig
+ *  signing key corresponding to the threshold policy.
+ *
+ *  Returns 1 on success, 0 if any input was invalid.
+ *
+ *  Args:         ctx: pointer to a context object initialized for signing and verification (cannot be NULL)
+ *            scratch: if non-NULL, pointer to a scratch space used to speed up computations
+ *  In/Out:    seckey: secret key belonging to the caller
+ *     signer_pubkeys: array of public keys corresponding to every signer (cannot be NULL)
+ *  In:        n_keys: number of keys in the above array
+ *         continuing: whether to initialize (zero) or update (nonzero) `seckey` and `signer_pubkeys`
+ *          privshard: if non-NULL, a private shard to verify against the computed public shards
+ *             my_idx: index of the caller in the MuSig policy
+ *          other_idx: index of the signer who provided the shards
+ *           pubcoeff: array of public coefficients (cannot be NULL)
+ *           n_coeffs: number of coefficients in the above array; or number of signers needed for quorum
+ */
+SECP256K1_API int secp256k1_thresholdsig_verify_shard(
+    const secp256k1_context *ctx,
+    secp256k1_scratch_space *scratch,
+    unsigned char *seckey,
+    secp256k1_pubkey *signer_pubkeys,
+    size_t n_keys,
+    const unsigned char *pk_hash,
+    int continuing,
+    const secp256k1_thresholdsig_keyshard *privshard,
+    size_t my_idx,
+    size_t other_idx,
+    const secp256k1_pubkey *pubcoeff,
+    size_t n_coeffs
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(4) SECP256K1_ARG_NONNULL(6) SECP256K1_ARG_NONNULL(11);
+
+/** Initializes a threshold signing session
```

Comment:
Would be good to mention that this is done once the participating signers are known. And that `signers` array refers to the participating signers.